src/callbacks.jl

```diff
@@ -67,6 +67,81 @@ function print_evaluation(results::Dict{String,Dict{String,Matrix{Float64}}}, it
 end
 
 
+function cb_save_evaluation(period::Int = 1, show_stdv::Bool = true, eval_save_file::AbstractString = "xgboost.log")
+    cb_timing = "after"
+
+    # Reset the file if it exists
+    write(eval_save_file, "")
+
+    function callback(env::CallbackEnv)
+        cb_timing
+
+        if env.rank != 0 || length(env.results) == 0
+            return nothing
+        end
+
+
+        iter = env.iteration
+        cur_results_idx = 1 + iter - env.begin_iteration
+        if (iter - 1) % period == 0 || iter == env.begin_iteration || iter == env.end_iteration
+            save_evaluation(env.results, iter, cur_results_idx, show_stdv, eval_save_file)
+        end
+
+        return nothing
+    end
+
+    return callback
+end
+
+
+function save_evaluation(results::Dict{String,Dict{String,Matrix{Float64}}}, iter::Int,
+                          cur_results_idx::Int, show_stdv::Bool, eval_save_file::AbstractString)
+
+    # create csv if it doesn't exist and write header
+    if filesize(eval_save_file) == 0
+        metrics_header = []
+        for eval_name in sort(collect(keys(results)))
```

Comment:
any reason this isn't a map? I'll admit I'm still getting the hang of julialang so I may just be missing something

Reply:
I think this could be a map. But I think iterating over dictionaries like this is more common / more human readable. 

I guess the bad about this is that I have to do this `sort(collect())` business as `keys()` aren't guaranteed to return in the same order (i.e. not doing sort collect could cause the columns to get mixed up).
Also, pushing onto `metrics` isn't ideal, but i couldn't work out a nice way to get the correct size.